src/Shared/PackageManagers/NuGetProjectManager.cs

```diff
@@ -104,93 +112,45 @@ public override async Task<IEnumerable<string>> DownloadVersion(PackageURL purl,
+
+                IPackageDownloader? downloader = await resource.GetPackageDownloaderAsync(
+                    packageIdentity,
+                    _sourceCacheContext,
+                    NullLogger.Instance, 
+                    cancellationToken);
+                
+                string? targetName = $"nuget-{packageName}@{packageVersion}";
+                string extractionPath = Path.Combine(TopLevelExtractionDirectory, targetName);
+                if (doExtract && Directory.Exists(extractionPath) && cached)
                 {
-                    if (catalogPage.TryGetProperty("items", out JsonElement itemElement))
-                    {
-                        foreach (JsonElement item in itemElement.EnumerateArray())
-                        {
-                            JsonElement catalogEntry = item.GetProperty("catalogEntry");
-                            string? version = catalogEntry.GetProperty("version").GetString();
-                            if (version != null && version.Equals(packageVersion, StringComparison.InvariantCultureIgnoreCase))
-                            {
-                                string? archive = catalogEntry.GetProperty("packageContent").GetString();
-                                HttpResponseMessage? result = await httpClient.GetAsync(archive);
-                                result.EnsureSuccessStatusCode();
-                                Logger.Debug("Downloading {0}...", purl?.ToString());
-
-                                string? targetName = $"nuget-{packageName}@{packageVersion}";
-                                string extractionPath = Path.Combine(TopLevelExtractionDirectory, targetName);
-                                if (doExtract && Directory.Exists(extractionPath) && cached == true)
-                                {
-                                    downloadedPaths.Add(extractionPath);
-                                    return downloadedPaths;
-                                }
+                    downloadedPaths.Add(extractionPath);
+                    return downloadedPaths;
+                }
 
-                                if (doExtract)
-                                {
-                                    downloadedPaths.Add(await ExtractArchive(targetName, await result.Content.ReadAsByteArrayAsync(), cached));
-                                }
-                                else
-                                {
-                                    targetName += Path.GetExtension(archive) ?? "";
-                                    await File.WriteAllBytesAsync(targetName, await result.Content.ReadAsByteArrayAsync());
-                                    downloadedPaths.Add(targetName);
-                                }
-                                return downloadedPaths;
-                            }
-                        }
+                
+                if (doExtract)
```

Comment:
Trying to understand the purpose of `doExtract` flag here as in when and how it's used by the caller.

If it's a much needed argument can't be removed, this if/else block can potentially be combined with the above if block where `doExtract` is true as well.

Reply:
Looks like the prior `if` statement block has additional conditions beyond just `doExtract`, so I'm not sure these can be directly merged @bhuwan-agarwal

Reply:
It looks to me like the second `if (doExtract)` maybe should be an else if to first.

Reply:
One option might be to move the `Directory.Exists(extractionPath) && cached` portion of the condition into a nested if statement here

Reply:
From poking around and testing, it's basically do you want to just download the `.nupkg` or do you want to download the `.nupkg` but extract the contents into a directory, leaving you with something like this:
![image](https://user-images.githubusercontent.com/8357054/158856822-0dce273d-57a7-4b21-9a48-8976c0a849d0.png)

Reply:
I wanted these two statements separated because if the first if statement, `if (doExtract && Directory.Exists(extractionPath) && cached)` evaluates to true, then the `resource.CopyNupkgToStreamAsync` doesn't need to be ran and we can return results earlier. It might not be a huge deal, but that's just how I was looking at it.